src/main/java/org/synyx/urlaubsverwaltung/person/extension/PersonEventHandlerExtension.java

```diff
@@ -0,0 +1,70 @@
+package org.synyx.urlaubsverwaltung.person.extension;
+
+import de.focus_shift.urlaubsverwaltung.extension.api.person.PersonCreatedEventDTO;
+import de.focus_shift.urlaubsverwaltung.extension.api.person.PersonDeletedEventDTO;
+import de.focus_shift.urlaubsverwaltung.extension.api.person.PersonDisabledEventDTO;
+import de.focus_shift.urlaubsverwaltung.extension.api.person.PersonUpdatedEventDTO;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+import org.synyx.urlaubsverwaltung.person.Person;
+import org.synyx.urlaubsverwaltung.person.PersonCreatedEvent;
+import org.synyx.urlaubsverwaltung.person.PersonDeletedEvent;
+import org.synyx.urlaubsverwaltung.person.PersonDisabledEvent;
+import org.synyx.urlaubsverwaltung.person.PersonService;
+import org.synyx.urlaubsverwaltung.person.PersonUpdatedEvent;
+
+@ConditionalOnProperty(value = "uv.extensions.enabled", havingValue = "true")
+@Component
+public class PersonEventHandlerExtension {
```

Comment:
Warum haben wir hier nicht ein Interface aus dem Extension package, welches wir ansprechen, ohne events, und dieses dann in der Extension implementieren? So wären die Wege synchron und wir könnten in der Extension schauen wie wir die Informationen weiter verarbeiten.

d.h. wir würden quasi ein "incomming" und ein "outgoing" interface anbieten und das incoming in der UV und das outgoing in der Extension impplementieren

Reply:
Verstehe den Sinn nicht, wieso die Extension das Interface implementieren soll, somit hätte die Extension eine Abhängigkeit auf die Domain-Events der Urlaubsverwaltung. Der Grund für die Extension-API ist genau der Grund losgelöst von Urlaubsverwaltunng-Code zu sein. D.h. die Urlaubsverwaltung hat die Verantwortung Domain-Objects entkoppelt als Extension-Events zu veröffentlichen